src/screens/Teachers.js

```diff
@@ -12,10 +13,30 @@ export default class TeachersScreen extends React.Component {
         this.state = {
             isLoading: true,
             data: props.route.params.teachers ?? [],
-            user: props.route.params.user ?? {}
+            user: props.route.params.user ?? {},
+            searchQuery: '',
         }
+        this.clearSearch = this.clearSearch.bind(this);
       }
 
+    handleSearch = (text) => {
+        this.setState({searchQuery: text})
+    }
+
+    clearSearch() {
+        this.setState({searchQuery: ''})
+    }
+
+    getFilteredData() {

```

Comment:
You don't need to bind this function?

Reply:
`getFilteredData()` is only called in this class's render method, so it should be good without the bind